File: LeetCode/Easy/1544-make-the-string-great/1544-make-the-string-great.java
class Solution {
    public String makeGood(String s) {
        Deque<Character> stack = new ArrayDeque<>();
        for (char c : s.toCharArray()) {
            if (!stack.isEmpty() && Math.abs(stack.peek() - c) == 32)
                stack.pop();
            else
                stack.push(c);
        }

        StringBuilder sb = new StringBuilder();
        while (!stack.isEmpty())
            sb.append(stack.pop());

        return sb.reverse().toString();
    }
}

File: LeetCode/Easy/1544-make-the-string-great/1544-make-the-string-great.kt
class Solution {
    fun makeGood(s: String): String {
        val sb = StringBuilder()

        for (c in s) {
            val last = sb.lastOrNull()

            if (last != null && abs(last - c) == 32)
                sb.deleteAt(sb.length - 1)
            else
                sb.append(c)
        }

        return sb.toString()
    }
}